libs/providers/multi-provider-web/src/lib/multi-provider-web.ts

```diff
@@ -232,32 +238,31 @@ export class WebMultiProvider implements Provider {
     hookContext: HookContext,
     hookHints: HookHints,
   ) {
-    let evaluationResult: ResolutionDetails<T>;
+    let evaluationDetails: EvaluationDetails<T>;
 
     try {
       this.hookExecutor.beforeHooks(provider.hooks, hookContext, hookHints);
 
-      evaluationResult = this.callProviderResolve(
+      const resolutionDetails = this.callProviderResolve<T>(
         provider,
         flagKey,
         defaultValue,
         hookContext.context,
       ) as ResolutionDetails<T>;
 
-      const afterHookEvalDetails = {
-        ...evaluationResult,
-        flagMetadata: Object.freeze(evaluationResult.flagMetadata ?? {}),
+      evaluationDetails = {
+        ...resolutionDetails,
+        flagMetadata: Object.freeze(resolutionDetails.flagMetadata ?? {}),
         flagKey,
       };
 
-      this.hookExecutor.afterHooks(provider.hooks, hookContext, afterHookEvalDetails, hookHints);
-      return evaluationResult;
+      this.hookExecutor.afterHooks(provider.hooks, hookContext, evaluationDetails, hookHints);
     } catch (error: unknown) {
       this.hookExecutor.errorHooks(provider.hooks, hookContext, error, hookHints);
-      throw error;
-    } finally {
-      this.hookExecutor.finallyHooks(provider.hooks, hookContext, hookHints);
+      evaluationDetails = this.getErrorEvaluationDetails(flagKey, defaultValue, error);
     }
+    this.hookExecutor.finallyHooks(provider.hooks, hookContext, evaluationDetails, hookHints);
+    return evaluationDetails;
```

Comment:
Beyond the naming, the important change here is that we return evaluation details in the finally hooks as well - that meant some changes to the flow here similar to what's done in the SDK.